QuizApp/NetworkServiceProtocol.swift

```diff
@@ -0,0 +1,5 @@
+import UIKit
+protocol NetworkServiceProtocol {
+    func executeUrlRequest<T: Decodable>(_ request: URLRequest, completionHandler: @escaping (Result<T, RequestError>) -> Void)
+    func executeUrlPostRequest(_ request: URLRequest, completionHandler: @escaping (ServerResponse) -> Void)
```

Comment:
The name is `executeUrlPostRequest` but you can pass any type of request as a parameter. Better to rename to something more explicit like `sendResult` and then create the URLRequest inside rather than passing it as a parameter. You would then need to pass `(startTime: Double, quizId: Int, correct: Int)` to this function (or even better a struct instance that encapsulates this data).
The final function definition might look like this:
`func sendResult(_ result: QuizResult, completionHandler: @escaping (ServerResponse) -> Void)`